### PR TITLE
plugin/decision: set config boundaries to upload_size_limit_bytes

### PR DESCRIPTION
**Pull request by sspaink**
### Why the changes in this PR are needed?

resolve: https://github.com/open-policy-agent/opa/issues/7564

### What are the changes in this PR?

Now there is a minimum and maximum number the user can set when configuring `upload_size_limit_bytes`.

* Minimum is 90 bytes
* Maximum is 4GB

If the value is configured out of these boundaries a warning is logged and the value is reset to the minimum or maximum:

```
{"level":"warning","msg":"the configured `upload_size_limit_bytes` (8223372036854776000) has been set to the maximum limit (4294967296)","plugin":"discovery","time":"2025-05-09T16:57:16-05:00"}
```

```
{"level":"warning","msg":"the configured `upload_size_limit_bytes` (-1) has been set to the minimum limit (90)","plugin":"discovery","time":"2025-05-09T16:58:20-05:00"}
```

### CODE REVIEW

File: v1/plugins/logs/plugin.go [hunk 1]
```diff
@@ -372,7 +374,20 @@ func (c *Config) validateAndInjectDefaults(services []string, pluginsList []stri
...
 
-	c.Reporting.UploadSizeLimitBytes = &uploadLimit
+	switch {
+	case uploadLimit > maxUploadSizeLimitBytes:
+		maxUploadLimit := maxUploadSizeLimitBytes
+		c.Reporting.UploadSizeLimitBytes = &maxUploadLimit
+		if l != nil {
+			l.Warn("the configured `upload_size_limit_bytes` (%d) has been set to the maximum limit (%d)", uploadLimit, maxUploadLimit)
+		}
+	case uploadLimit <= minUploadSizeLimitBytes:
```
Comment (johanfylling) [inline v1/plugins/logs/plugin.go:384]: if `uploadLimit == minUploadSizeLimitBytes`, won't that give us a pretty peculiar warning message?

```
the configured `upload_size_limit_bytes` (90) has been set to the minimum limit (90)
```

File: v1/plugins/logs/plugin_test.go [hunk 2]
```diff
@@ -3654,3 +3654,68 @@ func testStatus() *bundle.Status {
...
+			limit:         1000,
+			expectedLimit: 1000,
+		},
+		{
+			name:          "negative limit",
+			limit:         -1,
+			expectedLimit: minUploadSizeLimitBytes,
+			expectedLog:   "the configured `upload_size_limit_bytes` (-1) has been set to the minimum limit (90)",
+		},
+	}
```
Comment (johanfylling) [inline v1/plugins/logs/plugin_test.go:3683]: We should add test cases for when the configured limit is set to exactly the minimum and maximum allowed values.

Comment (sspaink) [reply v1/plugins/logs/plugin_test.go:3693]: good idea, test cases added

File: v1/plugins/logs/plugin.go (hunk 1 above)
Comment (sspaink) [reply v1/plugins/logs/plugin.go]: fixed and test case added

File: v1/plugins/logs/plugin.go [hunk 3]
```diff
@@ -372,7 +374,20 @@ func (c *Config) validateAndInjectDefaults(services []string, pluginsList []stri
...
+		minUploadLimit := minUploadSizeLimitBytes
+		c.Reporting.UploadSizeLimitBytes = &minUploadLimit
+		l.Warn("the configured `upload_size_limit_bytes` (%d) has been set to the minimum limit (%d)", uploadLimit, minUploadLimit)
```
[the remark this answers is not in the document]
Comment (sspaink) [reply v1/plugins/logs/plugin.go]: good catch, nil-check added

**Review by johanfylling — APPROVED** [review]
Thanks! 👍
